lib/still/preprocessor.ex

```diff
@@ -131,32 +131,48 @@ defmodule Still.Preprocessor do
   end
 
   defp do_run(file, [preprocessor | remaining_preprocessors]) do
-    preprocessor.run(file)
-    |> run(remaining_preprocessors)
-  catch
-    :error, %{description: description} when description != "" ->
-      raise PreprocessorError,
-        message: description,
+    try do
+      preprocessor.run(file)
+      |> run(remaining_preprocessors)
+    catch
+      :error,
+      %PreprocessorError{
+        message: message,
         preprocessor: preprocessor,
         remaining_preprocessors: remaining_preprocessors,
-        source_file: file,
-        stacktrace: __STACKTRACE__
-
-    :error, e ->
-      Logger.error(inspect(e))
-
-      case e do
-        %PreprocessorError{} ->
+        source_file: source_file,
+        stacktrace: stacktrace
+      } = e ->
+        if String.contains?(source_file.input_file, file.input_file) do
           raise e
-
-        e ->
+        else
           raise PreprocessorError,
-            message: inspect(e),
+            message: message,
             preprocessor: preprocessor,
             remaining_preprocessors: remaining_preprocessors,
-            source_file: file,
-            stacktrace: __STACKTRACE__
-      end
+            source_file: %{
+              source_file
+              | input_file: "#{file.input_file} -> #{source_file.input_file}"
```

Comment:
Would you consider adding a `compilation_chain` field instead of this?

Reply:
I thought about it but it doesn't solve the issue. The `input_file` is our unique identifier, if I don't change we won't be able to distinguish between errors in included files.

Reply:
Yeah but we can keep the `input_file` field and simply move this to a new one, right? Use that instead.

Reply:
You know what, this doesn't work. I need to rethink this feature.

Reply:
I'll try something else because this approach will not clear the errors properly.